lib/typings/src/entity/entity-repository.ts

```diff
@@ -4,14 +4,18 @@ import { Observable } from 'rxjs';
 import { EntityUpdate } from './entity-update';
 import { NgxsEntityCollections } from './ngxs-entity-collections';
 
-export interface EntityRepository<V, K extends string | number> {
+export interface EntityRepository<
+    V,
+    K extends string | number,
+    C extends NgxsEntityCollections<V, K> = NgxsEntityCollections<V, K>
```

Comment:
In fact, I propose a different model of behavior
what do you think about?

```ts
export interface EntityRepository<V, K extends string | number, C = {}> {}

getState(): NgxsEntityCollections<V, K, C>
```

```ts
export type NgxsEntityCollections<V, K extends number | string = EntityIdType, C = {}> = {
    ids: K[];
    entities: EntityDictionary<K, V>;
} & C;

export class NgxsDataEntityCollectionsRepository<V, K extends number | string = EntityIdType, C = {}>
    extends AbstractRepository<NgxsEntityCollections<V, K, C>>
    implements EntityRepository<V, K> {
}
```

And now

```ts
interface ArticleOptions {
       loading: boolean;
       category: string;	      
};

@StateRepository()
@State({
      name: 'articles',
      defaults: {
          ...createEntityCollections(),
          loading: false,
          loaded: false,
    },
})
@Injectable()
class ArticlesState extends NgxsDataEntityCollectionsRepository<Article, number, ArticleOptions> { }
```